pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -635,7 +635,7 @@ func (m *kubeGenericRuntimeManager) pruneInitContainersBeforeStart(pod *v1.Pod,
 	for name := range initContainerNames {
 		count := 0
 		for _, status := range podStatus.ContainerStatuses {
-			if status.Name != name || !initContainerNames.Has(status.Name) ||
```

Comment:
@yjhong looks like you wrote this originally. What was the reason for checking `initContainerNames.Has` here? (naively looking it it, we're already iterating the names and statuses in this nested loop, so I'm not sure why we'd iterate the names again)